src/util/performance.cpp

```diff
@@ -267,114 +279,83 @@ void report() {
     style.PlotPadding = Vec2();
     style.PlotBorderSize = 0.0;
     ImPlot::SetNextPlotLimits(0, HISTORY_LENGTH, 0, MAXIMUM_MS);
-    if (ImPlot::BeginPlot("##FrameTimes",
-                          "Total Frame Time (ms)",
-                          "",
-                          Vec2(-1, GRAPH_HEIGHT),
-                          ImPlotFlags_None,
-                          ImPlotAxisFlags_Lock | ImPlotAxisFlags_NoDecorations,
-                          ImPlotAxisFlags_Lock)) {
-        ImPlot::SetLegendLocation(ImPlotLocation_North, ImPlotOrientation_Horizontal, true);
-        DRAW_NOW_LINE;
-        ImPlot::PlotLine("Raw", frame_time, HISTORY_LENGTH);
 
-        const u32 SAMPLES_IN_AVERAGE = Math::min<u32>(30, HISTORY_LENGTH);
-        u32 start_index = (HISTORY_LENGTH + frame_index - SAMPLES_IN_AVERAGE) % HISTORY_LENGTH;
+    // Utility function for drawing a horizontal line for the
+    // specified frame.
+    auto draw_frame_line = [](i32 frame) {
+        f32 xs[] = { (f32)frame, (f32)frame };
+        f32 ys[] = { -10, 30 };
+        ImPlot::PlotLine("Now", xs, ys, 2);
+    };
+
+    // Averages the time per frame for a cirtain history.
+    auto calculate_average_frame_time = [](i32 frame, i32 lookback) {
+        const u32 num_samples = Math::min<u32>(lookback, HISTORY_LENGTH);
+        u32 start_index = (HISTORY_LENGTH + frame - num_samples) % HISTORY_LENGTH;
         f32 average = 0.0;
-        for (u32 i = 0; i < SAMPLES_IN_AVERAGE; i++) {
+        for (u32 i = 0; i < num_samples; i++) {
             u32 sample_index = (start_index + i) % HISTORY_LENGTH;
             average += frame_time[sample_index];
         }
-        average /= SAMPLES_IN_AVERAGE;
-
-        f32 xs[] = { -100, (f32)start_index, (f32)frame_index, HISTORY_LENGTH };
+        average /= num_samples;
+        return average;
+    };
+
+    // The length of previous frames, and the average frame time.
+    ImGui::Text("> Total Frame Time");
+    if (ImPlot::BeginPlot("##???I'm Confused",
```

Comment:
What's up here?

Reply:
I mixed up the title and the x-axis label. And then I just left that there... Because... It works... xD